src/components/AddExpensesForm.js

```diff
@@ -0,0 +1,91 @@
+import { useState } from 'react';
+import { v4 as uuid } from 'uuid';
+
+const AddExpensesForm = () => {
+  const initialState = {
+    id: '',
+    name: '',
+    date: '',
+    amount: '',
+  };
+
+  const [singleExpense, setSingleExpense] = useState(initialState);
+  const [expenses, setExpenses] = useState([]);
+
+  const handleChange = (evt) => {
+    const { name, value } = evt.target;
+    setSingleExpense({
+      ...singleExpense,
+      [name]: value,
+    });
+  };
+
+  const handleSubmit = (evt) => {
+    evt.preventDefault();
+    const newExpenses = [...expenses, { ...singleExpense, id: uuid() }];
+    setExpenses(newExpenses);
+    setSingleExpense(initialState);
+  };
+
+  return (
+    <div className="container-md">
+      <div className="text-center">
+        <h2 className="display-2">Add some Expenses</h2>
+      </div>
+      <form className="row g-2" onSubmit={handleSubmit}>
+        <div className="col-md">
+          <div className="form-floating">
+            <input
+              className="form-control"
+              type="text"
+              id="name"
+              name="name"
+              required
+              onChange={handleChange}
+              value={singleExpense.name}
+              aria-label="Expense Name"
+              placeholder="Expense Name"
+            />
+            <label htmlFor="amount">Expense Name</label>
+          </div>
+        </div>
+        <div className="col-md">
```

Comment:
what does `col-md` do differently than `col`?

Reply:
can you please point me to the documentation of this feature? I only know `col-md-*`

Reply:
Sure! here it is: https://getbootstrap.com/docs/5.1/forms/floating-labels/#layout

Reply:
you are completely right this is great news. this feature with just using `col-md` is basically undocumented. thanks for sharing

Reply:
🏆